app/controllers/icon_redirects_controller.rb

```diff
@@ -1,9 +1,11 @@
 class IconRedirectsController < ApplicationController
   def show
+    response.headers["Cache-Control"] = "max-age=315360000, public, immutable"
```

Comment:
I don't think think these are quite the right headers as these icons may change over time - the immutable indefinitely cacheable resource is the item that is redirected too.

Otherwise if the icon changes caches for www.gov.uk/apple-touch-icon-114x114-precomposed.png will indefinitely redirect to an old resource.

I'd suggest going for a more modest time like 1 hour? You kind of want it to be long enough to have an impact but be short enough we don't actually want to have to ever purge the cache when things change.

You can also set it in a nicer Rails fashion with something like `expires_in(1.hour, public: true)` if you put it in a before action you could specify once for the whole controller. For example:  `before_action { expires_in(1.hour, public: true) }`

Reply:
This is the header for the redirect, not the actual asset we redirect to. We're essentially making the redirect permanent. The assets themselves have a different header.

Reply:
Hmm I feel like one of us has our wires crossed. Apologies if that ends up being me!

This is the action for a request to https://www.gov.uk/favicon.ico right?

So currently when we send a request to https://www.gov.uk/favicon.ico we receive a redirect to https://www.gov.uk/assets/static/favicon-8d811b8c3badbc0b0e2f6e25d3660a96cc0cca7993e6f32e98785f205fc40907.ico and that has a cache header of "cache-control: no-cache"

With your change a request to https://www.gov.uk/favicon.ico would return a redirect to https://www.gov.uk/assets/static/favicon-8d811b8c3badbc0b0e2f6e25d3660a96cc0cca7993e6f32e98785f205fc40907.ico which should be cached for 10 years?

Then imagine if we change the favicon asset at some point - a new asset would be created with a new fingerprint id and that 10 year cache redirect would at best be going to the old item and at worse be going to a 404 (given we don't have reason to keep old asset versions forever).

Reply:
Sorry I replied before your edit seems like we were more in alignment than I realised!

Reply:
Reduced to 1 day and used the more elegant `before_action` method. The actual assets we point to are set to 10 years – tried to be consistent with our current caching policy 🤷‍♂️

Reply:
The actual assets are fine to be 10 years because we'd never expect https://www.gov.uk/assets/static/favicon-8d811b8c3badbc0b0e2f6e25d3660a96cc0cca7993e6f32e98785f205fc40907.ico to return any other response than that version of the asset. Whereas we'd expect www.gov.uk/favico.ico to vary based on what the current favicon is.

Reply:
To me `https://www.gov.uk/assets/static/favicon-8d811b8c3badbc0b0e2f6e25d3660a96cc0cca7993e6f32e98785f205fc40907.ico` returns `max-age=315360000, public, immutable`